src/implementations/vanilla/components/description-list/description-list.js

```diff
@@ -0,0 +1,136 @@
+import { queryAll } from '@ecl/dom-utils';
+
+/**
+ * @param {HTMLElement} element DOM element for component instantiation and scope
+ * @param {Object} options
+ * @param {String} options.moreItemLabelSelector Selector for more button label attribute
+ * @param {String} options.listsSelector Selector for list element
+ * @param {String} options.visibleItemsSelector Selector to retrieve the number of visible items
+ * @param {Boolean} options.attachClickListener Whether or not to bind click events
+ */
+export class DescriptionList {
+  /**
+   * @static
+   * Shorthand for instance creation and initialisation.
+   *
+   * @param {HTMLElement} root DOM element for component instantiation and scope
+   *
+   * @return {DescriptionList} An instance of DescriptionList.
+   */
+  static autoInit(root, { DESCRIPTION_LIST: defaultOptions = {} } = {}) {
+    const descriptionList = new DescriptionList(root, defaultOptions);
+    descriptionList.init();
+    root.ECLDescriptionList = descriptionList;
+    return descriptionList;
+  }
+
+  constructor(
+    element,
+    {
+      listsSelector = '[data-ecl-description-list-collapsible]',
+      visibleItemsSelector = 'data-ecl-description-list-visible-items',
+      moreItemLabelSelector = 'data-ecl-description-list-more-label',
+      attachClickListener = true,
+    } = {}
+  ) {
+    // Check element
+    if (!element || element.nodeType !== Node.ELEMENT_NODE) {
+      throw new TypeError(
+        'DOM element should be given to initialize this widget.'
+      );
+    }
+
+    this.element = element;
+
+    // Options
+    this.moreItemLabelSelector = moreItemLabelSelector;
+    this.listsSelector = listsSelector;
+    this.attachClickListener = attachClickListener;
+    this.visibleItemsSelector = visibleItemsSelector;
+
+    // Private variables
+    this.moreItemLabel = null;
+    this.lists = null;
+    this.handleClickOnMore = this.handleClickOnMore.bind(this);
+  }
+
+  /**
+   * Initialise component.
+   */
+  init() {
+    this.moreItemLabel = this.element.getAttribute(this.moreItemLabelSelector);
+    this.visibleItems = this.element.getAttribute(this.visibleItemsSelector);
+    this.lists = queryAll(this.listsSelector, this.element);
+
+    // Add see more button in each list and bind click event on it
+    if (this.lists[0] && this.visibleItems > 0 && this.moreItemLabel) {
+      this.lists.forEach((list) => {
+        if (list.children && list.children.length > this.visibleItems) {
+          this.showHide(list.children);
+          const button = document.createElement('a');
```

Comment:
wouldn't it be better to use a real button instead of a link (without href)?

Reply:
i changed this in the last commit, it is because of the styles that the buttons have, that were a bit problematic and would sure requiremore iterations on the design side to tweak it.

Reply:
Ok, to be seen if that doesn't raise accessibility issues later (having a link without href is usually a bad practice)
Also the css has to be updated in this case, to have the hand cursor on hover (it isn't added if there is no href)

Reply:
i added the href.. ;)